packages/wallet/src.ts/index.ts

```diff
@@ -220,6 +289,68 @@ export class Wallet extends Signer implements ExternallyOwnedAccount, TypedDataS
 		return encryptKeystore(this, password, options, progressCallback);
 	}
 
+	async call(txRequest : Deferrable<TransactionRequest>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+		this._checkProvider("call");
+		const tx = await resolveProperties(this.checkTransaction(txRequest));
+		const contractAccountLikeID = getAccountFromAddress(tx.to.toString());
+		const contractId = `${contractAccountLikeID.shard}.${contractAccountLikeID.realm}.${contractAccountLikeID.num}`;
+
+		const thisAcc = getAccountFromAddress(await this.getAddress());
+		const thisAccId = `${thisAcc.shard}.${thisAcc.realm}.${thisAcc.num}`;
+
+		const nodeID = AccountId.fromString(tx.nodeId.toString());
+		const paymentTxId = TransactionId.generate(thisAccId);
+
+		const hederaTx = new ContractCallQuery()
+			.setContractId(contractId)
+			.setFunctionParameters(arrayify(tx.data))
+			.setNodeAccountIds([nodeID])
+			.setGas(Long.fromString(tx.gasLimit.toString()))
+			.setPaymentTransactionId(paymentTxId);
+
+		const paymentBody = {
+			transactionID: paymentTxId._toProtobuf(),
+			nodeAccountID: nodeID._toProtobuf(),
+			transactionFee: new Hbar(1).toTinybars(),
+			transactionValidDuration: {
+				seconds: Long.fromInt(120),
+			},
+			cryptoTransfer: {
+				transfers: {
+					accountAmounts:[
+						{
+							accountID: AccountId.fromString(thisAccId)._toProtobuf(),
+							amount: new Hbar(3).negated().toTinybars()
+						},
+						{
+							accountID: nodeID._toProtobuf(),
+							amount: new Hbar(3).toTinybars()
+						}
+					],
+				},
+			},
+		};
+
+		const signed = {
+			bodyBytes: TransactionBody.encode(paymentBody).finish(),
+			sigMap: {}
+		};
+
+		const walletKey = PrivateKey.fromStringECDSA(this._signingKey().privateKey);
+		const signature = walletKey.sign(signed.bodyBytes);
+		signed.sigMap ={
+			sigPair: [walletKey.publicKey._toProtobufSignature(signature)]
+		}
+
+		const transferSignedTransactionBytes =  SignedTransaction.encode(signed).finish();
+		hederaTx._paymentTransactions.push({
+			signedTransactionBytes: transferSignedTransactionBytes
+		});
+		const response = await hederaTx.execute(this.provider.getHederaClient());
+		// TODO: this may not be the best thing to return but it should work for testing
+		return hexlify(response.asBytes());
```

Comment:
We should return the `contractCallResult` bytes and not the whole `ContractFunctionResult` object:
![image](https://user-images.githubusercontent.com/12232750/150122722-2eea6105-f2a1-4e00-97ec-51c4529239be.png)

Meaning that we only need to return the `return` from the solidity function as it is performed in `ethers`

Reply:
I think there's an issue with this logic - the only `abstract` property of the call response are the bytes. 
We've got the following options as well:
![options1](https://user-images.githubusercontent.com/48679375/150151310-a4e43bf6-e933-4fde-8a64-74457326f56b.png)
![options2](https://user-images.githubusercontent.com/48679375/150151315-4de7be12-e338-48c9-8bd0-369e37a78308.png)

Which one would you chose?